File: javascript_exercises/longest_word.js
// Longest Word

// Have the function LongestWord(sen) take the sen parameter being passed and return the largest word
// in the string. If there are two or more words that are the same length, return the first word from
// the string with that length. Ignore punctuation and assume sen will not be empty.

function LongestWord(sen) {

  var words = sen.replace(/[^a-zA-Z\s]/g, '').split(' ');
  var LongestWord = words.reduce ( function( longestSoFar, currentWord) {

    return currentWord.length > longestSoFar.length ? currentWord : longestSoFar;
  }, '');

  return LongestWord;
}

LongestWord('Argument goes here');

// Argument

File: ruby_exercises/longest_word.rb
# Longest Word

# Using the Ruby language, have the function LongestWord(sen) take the sen parameter being passed
# and return the largest word in the string. If there are two or more words that are the same length,
# return the first word from the string with that length. Ignore punctuation and assume sen will not be empty.

def LongestWord(str)
  words = str.split.map { |s| s.gsub(/\W/, '') }
  longest = words [0]
  words.each { |word| longest = word if word.length > longest.length }
  longest
end

LongestWord("This random words should have capitals")

# capitals
